Bottom: 76117077ebd9ada8f4a1b61d2ba78e05914d6b0c
Top:    61bc3aa0cd7125b03e0e3c0b136f31004d57f6a6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-15 00:09:59 +0000

Refresh of make-rats-snap-to-the-center-o

---

diff --git a/src/netlist.c b/src/netlist.c
index 7cb923e..d9592b0 100644
--- a/src/netlist.c
+++ b/src/netlist.c
@@ -137,10 +137,8 @@ pin_name_to_xy (LibraryEntryType * pin, int *x, int *y)
       *y = ((PinType *) (conn.ptr2))->Y;
       return 0;
     case PAD_TYPE:
-      *x = (((PadType *) (conn.ptr2))->Point1.X +
-            ((PadType *) (conn.ptr2))->Point2.X) / 2;
-      *y = (((PadType *) (conn.ptr2))->Point1.Y +
-            ((PadType *) (conn.ptr2))->Point2.Y) / 2;
+      *x = ((PadType *) (conn.ptr2))->Point1.X;
+      *y = ((PadType *) (conn.ptr2))->Point1.Y;
       return 0;
     }
   return 1;
diff --git a/src/rats.c b/src/rats.c
index 85a3f93..35047b5 100644
--- a/src/rats.c
+++ b/src/rats.c
@@ -139,8 +139,16 @@ FindPad (char *ElementName, char *PinNum, ConnectionType * conn, Boolean Same)
 	    conn->ptr2 = &element->Pad[i];
 	    conn->group =
 	      TEST_FLAG (ONSOLDERFLAG, &element->Pad[i]) ? SLayer : CLayer;
-	      conn->X = (element->Pad[i].Point1.X + element->Pad[i].Point2.X) / 2;
-	      conn->Y = (element->Pad[i].Point1.Y + element->Pad[i].Point2.Y) / 2;
+	    if (TEST_FLAG (EDGE2FLAG, &element->Pad[i]))
+	      {
+		conn->X = element->Pad[i].Point2.X;
+		conn->Y = element->Pad[i].Point2.Y;
+	      }
+	    else
+	      {
+		conn->X = element->Pad[i].Point1.X;
+		conn->Y = element->Pad[i].Point1.Y;
+	      }
 	    break;
 	  }
       if (i == element->PadN)
